Clarifications on fragids. See #50. (#51)
ietf-wg-httpapi/mediatypes: draft-ietf-httpapi-yaml-mediatypes.md

alias nodes (see Section 3.2.2.2 and 7.1 of [YAML])
as fragment identifiers to designate nodes.

An empty fragment identifier reference the root
of the YAML representation graph.

> 

Reviewer: The problem really is that this section is entirely out of place -- why is this in a different place from the JSON Pointer part?
I'm not so hot for entirely defining the JSON Pointer part in the media type template.

> 

Author: @cabo here we define how to process fragment identifiers alias nodes: this is specific to YAML and can be reused by other specifications (e.g. xxx+yaml) that are not interested in JSON Pointer.

> 

Author: @eemeli 

> This seems out of place here, esp. given the immediately preceding paragraph?

Do you think it's out of place because alias nodes cannot be empty? The point is that 
fragment identification includes the ability to send an empty fragment, and in this case we need to specify
how to handle them.

> 

Reviewer: > @cabo here we define how to process fragment identifiers alias nodes: this is specific to YAML and can be reused by other specifications (e.g. xxx+yaml) that are not interested in JSON Pointer.

First of all, I'm strongly against being +yaml specs "not being interested in JSON Pointer".
+yaml should always include application/yaml fragment syntax (Liskov substitution is the whole point of sss).
But assuming that this is not so, why does an implementation have to be interested in anchors to get this essential JSON Pointer feature?  (Empty JSON Pointer = entire tree.)

> 

Reviewer 2: @ioggstream I have to agree with @cabo

You've explained why you did it this way, and I follow the steps in your reasoning.  But I still find all of it very confusing and counter-intuitive compared to just saying outright that you support RFC 6901 JSON Pointer fragments.  There's nothing stopping you from defining a fragment syntax that overlaps when it comes to the empty fragment, as long as the semantics are the same, but this weird carve-out-and-replacement makes things so much harder to understand.

Also agree with the Liskov substitution point.

> 

Reviewer: (And, of course, there never will be parity between the two media types.  
I'm not even sure what that would mean.)

> 

Reviewer 2: @cabo agreed, JSON's refusal to adopt the de-facto standard as an official standard is a huge pain.  You end up having to specify things as URI + (non-fragment) JSON Pointer pairs in order to address part of a JSON resource without violating the standards.

   (eg. wrt on identifying parsers or treat downloads)

Q: Why using alias nodes as fragment identifiers?

> :  Fragment identifiers in YAML always reference
>    YAML representation graph nodes.
>    Since YAML is not limited to string keywords,
>    it uses aliases to reference nodes.
>    Alias nodes are thus the natural choice for fragment identifiers
>    {{application-yaml-fragment}}.

Reviewer 3: ```suggestion
:  Alias nodes are a native YAML feature that allows
   addressing any node in a document.
   Since YAML is not limited to string keywords,
   not all nodes are addressable using JSON Pointers.
   Alias nodes are thus the natural choice for fragment identifiers
   {{application-yaml-fragment}}.
```

Q: Why not using plain names for alias nodes? Why not defining plain names?

> :  The first version of this document only defined plain name fragments
>    referencing named anchors.
>    This behavior could have limited the ability of future xxx+yaml
>    media types to define their plain names fragments.

Reviewer 3: ```suggestion
:  Using plain name fragments could have
   limited the ability of future xxx+yaml
   media types to define their plain names fragments.
```

Referring to a past unreleased version of this document isn't really useful to a new reader.

Author: @eemeli OK. I wrote it only to specify that we already thought about it, but I agree with you.
